tests/ui/traits/new-solver/structural-resolve-field.rs

```diff
@@ -1,13 +1,35 @@
 // compile-flags: -Ztrait-solver=next
 // check-pass
 
-#[derive(Default)]
 struct Foo {
     x: i32,
 }
 
+impl MyDefault for Foo {
+    fn my_default() -> Self {
+        Self {
+            x: 0,
+        }
+    }
+}
+
+trait MyDefault {
+    fn my_default() -> Self;
+}
+
+impl MyDefault for [Foo; 0]  {
+    fn my_default() -> Self {
+        []
+    }
+}
+impl MyDefault for [Foo; 1] {
+    fn my_default() -> Self {
+        [Foo::my_default(); 1]
+    }
+}
+
 fn main() {
-    let mut xs = <[Foo; 1]>::default();
```

Comment:
attemtping to prove `[Foo; 1]: Default` breaks under the new solver with this PR because the impl in std is `[Foo; AnonConst]: Default` which we equate with `[Foo; 1]: Default`. New solver does not normalize constants so `AnonConst` does not unify with `1` even though evaluating the anon const would result in `1`. This test was accidentally relying on the `super_relate_consts` hack which is a great example of why we need this PR lol